### ISSUE TITLE
Consolidate product configuration and operator framework

### ISSUE THREAD

**Issue by razvan** [stackabletech/issues#34]
The goal of this epic is to bundle existing work which is loosely coupled (or not coupled at all) across various branches and issues. Future work should be linked here and provide a clear description of the goal from a customer perspective. 

The technical goal is to advance the `product-config` and `operator-rs` framework to the point where they are solid and largely stable foundation for stackable operators.

## Acceptance criteria

(wip)
(create separate issues  for each)

- `product-config` is generalized to be usable across all operators without modification.
- operators should be largely agnostic to package versioning (remove hard-coded versions from the zookeeper-operator)

## Questions

- should `product-config` support some form of templating ?
  - Not at the moment.
- Where will the product-config database live and in what form? This is relevant to the questions like: should the existing json format be optimized for humans to read/write or not ?
  - The "gold standard" will be packaged for customers and distributed in form of one or more files. Open-source users will have access to a simple (non optimized)  package. 

## Dependency management

- operators depend solely on `product-config` for parameter configuration (for start commands and env variables as well, not just configmaps)
- only operator-rs depends on product config (rexports)
- only operator-rs depends on k8s-openapi and kube* (reexports)

**Comment by soenkeliebau** [comment]
Closing this for now, we will address this at some point, when we create a roadmap for the product config framework.
